filebeat/input/tcp/harvester_test.go

```diff
@@ -0,0 +1,266 @@
+package tcp
```

Comment:
Nitpicking, I am completely fine if you ignore this comment:
I have seen that you used top-down design, I am not sure if it was on purpose, it made reviewing your code much easier. Do you mind reordering the functions here, so it would follow the same pattern?